packages/panna-sdk/src/react/hooks/use-account-events.ts

```diff
@@ -0,0 +1,253 @@
+            social: {
+              type: 'email',
+              data: `wallet-${address.slice(-8)}@unknown.domain`
+            }
+          } as OnConnectActivityRequest;
+        } else {
+          payload = {
+            ...basePayload,
+            eventType: 'onConnect',
+            smartAccount: transformSmartAccount(smartAccountConfig),
+            social: socialInfo
+          } as OnConnectActivityRequest;
+        }
+      } else if (eventType === 'disconnect') {
+        payload = {
+          ...basePayload,
+          eventType: 'disconnect',
+          ...(eventOptions.reason && { reason: eventOptions.reason })
+        } as DisconnectActivityRequest;
+      } else if (eventType === 'accountUpdate') {
+        payload = {
+          ...basePayload,
+          eventType: 'accountUpdate',
+          ...(eventOptions.updateType && {
+            updateType: eventOptions.updateType
+          })
+        } as AccountUpdateActivityRequest;
+      } else {
+        throw new Error(`Unsupported event type: ${eventType}`);
+      }
+
+      await pannaApiService.sendAccountEvent(address, payload, authToken);
+    } catch (error) {
+      console.error(`Failed to send ${eventType} event:`, error);
+    }
+  };
+
+  /**
+   * Handle wallet onConnect event
+   */
+  const handleOnConnect = async () => {
+    if (!userAddress) return;
+
+    try {
+      let socialInfo:
+        | { type: 'email' | 'phone' | 'google'; data: string }
+        | undefined;
+
+      // Try to get social profile information if available using core functions
+      try {
+        const { client, partnerId: currentPartnerId } = usePanna();
+
+        if (client && currentPartnerId) {
+          const ecosystemConfig = {
+            id: EcosystemId.LISK,
+            partnerId: currentPartnerId
+          };
+
+          try {
+            const email = await getEmail({
+              client,
+              ecosystem: ecosystemConfig
+            });
+            if (email) {
+              socialInfo = {
+                type: 'email',
+                data: email
+              };
+            }
+          } catch (error) {
+            console.debug('Could not get user email:', error);
+          }
+
+          if (!socialInfo) {
+            try {
+              const phone = await getPhoneNumber({
+                client,
+                ecosystem: ecosystemConfig
+              });
+              if (phone) {
+                socialInfo = {
+                  type: 'phone',
+                  data: phone
+                };
+              }
+            } catch (error) {
+              console.debug('Could not get user phone:', error);
+            }
+          }
+        }
+      } catch (error) {
+        // Social profile information is optional
+        console.debug('Could not retrieve social profile information:', error);
+      }
+
+      await sendAccountEvent('onConnect', userAddress, { social: socialInfo });
```

Comment:
we are also sending `sendAccountEvent` in `account-event-provider.ts` I don;t see much relation between them and also the one here doesn't gets triggered when connect or disconnect happens